File: xx_docs/selenium/1 - Selenium Testing Tools Cookbook [arquivo-apoio]/Chapter 2/Ruby/element_test.rb
require 'rubygems'
require 'selenium-webdriver'
require 'test/unit'

class ElementTest < Test::Unit::TestCase
	def setup
		@driver = Selenium::WebDriver.for :chrome
		@driver.get 'http://dl.dropbox.com/u/55228056/DoubleClickDemo.html'
		@verification_errors = []
	end
	
	def test_element_text
		#Get the message Element
		message = @driver.find_element :id => "message"
				
		#Verify message element's text displays "Click on me and my color will change"
		assert_equal "Click on me and my color will change", message.text
		
		#Get the area Element
		area = @driver.find_element :id => "area"
		
		#Verify area element's text displays "Div's Text\nSpan's Text"
		assert_equal "Div's Text\nSpan's Text",area.text
	end
	
	def test_element_attribute
		#Get the message Element
		message = @driver.find_element :id => "message"
		assert_equal "justify", message.attribute("align")
	end
	
	def test_element_style
		#Get the message Element
		message = @driver.find_element :id => "message"
		assert_equal "150px", message.css_value("width")
	end
	
	def teardown
		@driver.quit
		assert_equal [], @verification_errors
	end
end

File: XX - Backup Project/xx_docs/selenium/1 - Selenium Testing Tools Cookbook [arquivo-apoio]/Chapter 2/Python/element_test.py
from selenium import webdriver
from selenium.webdriver.common.action_chains import ActionChains
from array import *
import time, unittest

class ElementTest (unittest.TestCase):
	def setUp(self)	:
		self.driver = webdriver.Chrome()
		self.driver.get("http://dl.dropbox.com/u/55228056/DoubleClickDemo.html")
		
	def test_element_text(self):
		driver = self.driver
		
		#Get the message Element
		message = driver.find_element_by_id("message")
				
		#Verify message element's text displays "Click on me and my color will change"
		self.assertEqual("Click on me and my color will change", message.text)
		
		#Get the area Element
		area = driver.find_element_by_id("area")
		
		#Verify area element's text displays "Div's Text\nSpan's Text"
		self.assertEqual("Div's Text\nSpan's Text", area.text)
	
	def test_element_attribute(self):
		driver = self.driver
		
		#Get the message Element
		message = driver.find_element_by_id("message")
		self.assertEqual("justify", message.get_attribute("align"))
	
	def test_element_style(self):
		driver = self.driver
		
		#Get the message Element
		message = driver.find_element_by_id("message")
		self.assertEqual("150px", message.value_of_css_property("width"))
		
	def tearDown(self):
		self.driver.close()
		
if __name__ == "__main__":
	unittest.main()